src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -524,6 +527,7 @@ export class ConnectionDialogService implements IConnectionDialogService {
 			Note: A self-signed certificate offers only limited protection and is not a recommended practice for production environments. Do you want to enable 'Trust server certificate' on this connection and retry? `);
 			let readMoreLink = "https://learn.microsoft.com/sql/database-engine/configure-windows/enable-encrypted-connections-to-the-database-engine"
 			actions.push(new Action('trustServerCert', enableTrustServerCert, undefined, true, async () => {
+				this._telemetryService?.createActionEvent(TelemetryKeys.TelemetryView.ConnectionDialog, TelemetryKeys.TelemetryAction.EnableTrustServerCertificate)
```

Comment:
1. No need for the ? - the service should always exist
2. Use `sendActionEvent` (or call .send() on the event) otherwise it'll never get sent

You also may want to consider switching this to track every time the user is shown the dialog and then include in it a property that indicates whether they enabled TrustServerCertificate so you can see what percentage of people see this and act on it without needing to try and correlate separate events.

Reply:
Error Event will be captured with connection completion, and that would contain **error code**, so we can use that information to measure number of users who received that error message dialog, while this count can tell how many of those users actually clicked on Enable Trust Server Certificate.

cc @erinstellato-ms